src/Product/WCProductAdapter.php

```diff
@@ -296,6 +296,10 @@ function ( $gallery_image_id ) use ( $image_size ) {
 		);
 		// Uniquify the set of additional images
 		$gallery_image_links = array_unique( $gallery_image_links, SORT_REGULAR );
+		
+		// Limit addtional image link up to 10
+		$gallery_image_links = array_slice( $gallery_image_links, 0, 10);
```

Comment:
Minor code style issue:

```suggestion
		$gallery_image_links = array_slice( $gallery_image_links, 0, 10 );
```